File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int start = 0;
        int end = numbers.length - 1;

        while (start < end) {
            if (start > 0 && numbers[start] == numbers[start - 1]) {
                start++;
            } else if (end < numbers.length - 1 && numbers[end] == numbers[end + 1]) {
                end--;
            } else {
                if (numbers[start] + numbers[end] == target) {
                    return new int[]{start + 1, end + 1};
                } else if (numbers[start] + numbers[end] > target) {
                    end--;
                } else {
                    start++;
                }                
            }
        }
        return new int[]{-1, -1};
    }
}

File: Data Structures & Algorithms/search-2d-matrix/submission-2.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        for i in range(len(matrix)):
            start = 0
            end = len(matrix[0])
            while start <= end:
                mid = (start + end) // 2
                if mid > len(matrix[0]) - 1:
                    break
                if matrix[i][mid] == target:
                    return True
                elif matrix[i][mid] > target:
                    end -= 1
                else:
                    start += 1
        
        return False